src/components/UriViewer/UriViewer.js

```diff
@@ -56,28 +64,70 @@ export const UriViewer = _.flow(
         setMetadata(metadata)
       }
     } catch (e) {
-      setLoadingError(await e.json())
+      // azure blob storage api only returns responses in xml format at this time
+      // https://feedback.azure.com/d365community/idea/0646f944-3a25-ec11-b6e6-000d3a4f0f84
+      if (isAzureUri(uri)) {
+        setLoadingError(await e.text())
+      } else {
+        setLoadingError(await e.json())
+      }
     }
   }
+
   useOnMount(() => {
     loadMetadata()
   })
 
   const { size, timeCreated, updated, bucket, name, fileName, accessUrl } = metadata || {}
   const gsUri = `gs://${bucket}/${name}`
   const downloadCommand = getDownloadCommand(fileName, gsUri, accessUrl)
-  if (isAzureUri(uri)) {
+
+  if (isAzureUri(uri) && azureStorage !== undefined) {
+    const { azureSasStorageUrl, fileName, lastModified, size } = azureStorage
+    uri = azureSasStorageUrl || uri
     return h(Modal, {
       onDismiss,
       title: 'File Details',
       showCancel: false,
       showX: true,
       okButton: 'Done'
-    },
-    [els.cell([
-      els.label('Filename'),
-      els.data(_.last(uri.split('/')).split('.').join('.\u200B')) // allow line break on periods
-    ]), div({ style: { marginTop: '2rem', fontSize: 14 } }, ['Download functionality for Azure files coming soon'])])
+    }, [
+      Utils.cond(
+        [loadingError, () => h(Fragment, [
```

Comment:
If there is only one element (the div in this case), then it doesn't need to be wrapped in a Fragment.